libarchive.cmake

```diff
@@ -0,0 +1,61 @@
+#
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
+
+if(WIN32)
+    set(CMAKE_PREFIX_PATH "${CMAKE_SOURCE_DIR}/prebuild/libarchive/")
+endif()
+
+find_package(LibArchive QUIET)
+
+if(NOT LibArchive_FOUND)
+# Run the build script for libarchive
+    if(WIN32)
+        add_custom_command(
+            COMMAND ${CMAKE_SOURCE_DIR}/scripts/build_libarchive.bat
```

Comment:
If `build_libarchive.bat` can build libarchive, why do we need the prebuilt version?